discovery/consul/consul_test.go

```diff
@@ -46,3 +51,35 @@ func TestNonConfiguredService(t *testing.T) {
 		t.Errorf("Expected service %s to be watched", "nonConfiguredServiceName")
 	}
 }
+
+func TestConsulDiscovery(t *testing.T) {
+	srv1, err := testutil.NewTestServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv1.Stop()
+
+	srv1.AddService(t, "testService", structs.HealthPassing, []string{"master"})
+	srv1.AddCheck(t, "service:testService", "testService", structs.HealthPassing)
+
+	conf := &config.ConsulSDConfig{
+		Server:       srv1.HTTPAddr,
+		Services:     []string{"testService"},
+		WatchTimeout: 10 * time.Millisecond,
+	}
+	consulDiscovery, err := NewDiscovery(conf)
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Millisecond*10)
```

Comment:
If I understand correctly, this basically means that you time out the SD after 10 milliseconds, after which it would report "Could not retrieve target group from embedded consul". 10 milliseconds is a bit short and could lead to a flaky test. Do we need the timeout here at all? In the broken case, we can otherwise just let the test as a whole time out.

Could you also clarify what you need the custom `WatchTimeout` here? Isn't that just the long-poll duration during which it keeps a connection open to Consul to be informed of new events? But if it gets the new service before that duration, it will not block anything? I'm not a Consul expert though.